### ISSUE TITLE
False positive E1010 AWS::Serverless::Function Version.FunctionArn not allowed for AWS::CloudFront::Distribution DistributionConfig.DefaultCacheBehavior.LambdaFunctionAssociations[].LambdaFunctionARN

### ISSUE THREAD

**Issue by dmarinuswoodwing** [aws-cloudformation/cfn-lint#4644]
### CloudFormation Lint Version

cfn-lint 1.54.0

### What operating system are you using?

Fedora 44

### Describe the bug

The attached template returns:
```
E1010 'Version.FunctionArn' is not one of ['SnapStartResponse', 'SnapStartResponse.ApplyOn', 'SnapStartResponse.OptimizationStatus', 'Arn'] in ['us-east-1', 'eu-west-1']
template.yaml
```

This seems to be introduced since v1.54.0 (v1.53.3 doesn't error on this)

### Expected behavior

No errors

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'

Resources:
  Lambda:
    Type: AWS::Serverless::Function

  CloudFront:
    Type: AWS::CloudFront::Distribution
    Properties:
      DistributionConfig:
        DefaultCacheBehavior:
          ViewerProtocolPolicy: redirect-to-https
          TargetOriginId: Dummy
          LambdaFunctionAssociations:
            - EventType: 'origin-request'
              IncludeBody: false
              LambdaFunctionARN: !GetAtt Lambda.Version.FunctionArn
        Enabled: 'true'
```
